models/articles.model.js

```diff
@@ -1,11 +1,26 @@
 const db = require('../db/connection.js');
 
-exports.returnAllArticles = () => {
-    const articlesQuery = 'SELECT articles.*, COUNT(comments.comment_id) as comment_count FROM articles LEFT JOIN comments ON comments.article_id = articles.article_id GROUP BY articles.article_id ORDER BY articles.created_at DESC;'
-    return db.query(articlesQuery).then(({ rows }) => {
+exports.returnAllArticles = (topic, sort_by, order) => {
+    let articlesQuery = 'SELECT articles.*, COUNT(comments.comment_id) as comment_count FROM articles LEFT JOIN comments ON comments.article_id = articles.article_id'
+
+    if(topic !== null) articlesQuery += ' WHERE topic = $1'
+    articlesQuery += ' GROUP BY articles.article_id'
+
+    const acceptableSort_By = ["title", "topic", "author", "body", "created_at", "article_img_url", "comment_count"]
+    if(!acceptableSort_By.includes(sort_by)) return Promise.reject({status:404, msg:"Incorrect sort_by"})
+    else {
+        articlesQuery += ` ORDER BY ${sort_by}`
+    }
+    if(order.toLowerCase() !== "desc" && order.toLowerCase() !== "asc") return Promise.reject({status:404, msg:"Incorrect order"})
```

Comment:
I might have kept up the same approach as with your sort_by with the greenlist for this. It allows an easier way to scale up your model.